common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -246,27 +252,32 @@ void toAvailable() {
 
     private static final class LocalPool<T> {
         private final int ratioInterval;
-        private final Queue<DefaultHandle<T>> pooledHandles;
+        private final MessagePassingQueue<DefaultHandle<T>> pooledHandles;
         private int ratioCounter;
 
+        @SuppressWarnings("unchecked")
         LocalPool(int maxCapacity, int ratioInterval, int chunkSize) {
             this.ratioInterval = ratioInterval;
-            pooledHandles = PlatformDependent.newMpscQueue(chunkSize, maxCapacity);
+            if (BLOCKING_POOL) {
+                pooledHandles = new BlockingMessageQueue<DefaultHandle<T>>(maxCapacity);
+            } else {
+                pooledHandles = (MessagePassingQueue<DefaultHandle<T>>) newMpscQueue(chunkSize, maxCapacity);
+            }
             ratioCounter = ratioInterval; // Start at interval so the first one will be recycled.
         }
 
         DefaultHandle<T> claim() {
-            Queue<DefaultHandle<T>> pooledHandles = this.pooledHandles;
+            MessagePassingQueue<DefaultHandle<T>> pooledHandles = this.pooledHandles;
             DefaultHandle<T> handle;
             do {
-                handle = pooledHandles.poll();
+                handle = pooledHandles.relaxedPoll();
             } while (handle != null && !handle.availableToClaim());
             return handle;
         }
 
         void release(DefaultHandle<T> handle) {
             handle.toAvailable();
-            pooledHandles.offer(handle);
+            pooledHandles.relaxedOffer(handle);
```

Comment:
Relaxed offer is used just for completeness but TBH mpsc doesn't do anything different from offer here iirc

Reply:
Yeah, this is just pulling in what #11960 is doing.

Reply:
@franz1981 while it does nothing special atm it could at a later point in time so I think it is best to just use it and get the benefit for free if it will do anything special in the future :)